tests/integration/tests/graph_test.go

```diff
@@ -0,0 +1,265 @@
+	assertGraph(params, assert)
+}
+
+func TestWorkloadRequestsPerSecond(t *testing.T) {
+	assert := assert.New(t)
+	params := map[string]string{"graphType": "workload", "edges": "requestsPerSecond", "duration": "21600s"}
+	assertGraph(params, assert)
+}
+
+func TestWorkloadRequestPercentage(t *testing.T) {
+	assert := assert.New(t)
+	params := map[string]string{"graphType": "workload", "edges": "requestsPercentage", "duration": "21600s"}
+	assertGraph(params, assert)
+}
+
+func TestWorkloadRequestsResponseTime(t *testing.T) {
+	assert := assert.New(t)
+	params := map[string]string{"graphType": "workload", "edges": "responseTime", "duration": "21600s"}
+	assertGraph(params, assert)
+}
+
+func TestWorkloadAppendersBlank(t *testing.T) {
+	assert := assert.New(t)
+	params := map[string]string{"graphType": "workload", "duration": "60s", "appenders": ""}
+	assertGraph(params, assert)
+}
+
+func TestBoxNegative(t *testing.T) {
+	assert := assert.New(t)
+	params := map[string]string{"graphType": "app", "duration": "60s"}
+	params["boxBy"] = "junk"
+	assertGraphInvalid(params, assert)
+}
+
+func TestGraphTypeNegative(t *testing.T) {
+	assert := assert.New(t)
+	params := map[string]string{"graphType": "app", "duration": "60s"}
+	params["graphType"] = "junk"
+	assertGraphInvalid(params, assert)
+}
+
+func TestDisplayIdleEdges(t *testing.T) {
+	assert := assert.New(t)
+	params := map[string]string{"graphType": "versionedApp", "duration": "300s", "idleEdges": "true"}
+	assertGraph(params, assert)
+}
+
+func TestDisplayIdleNodes(t *testing.T) {
+	assert := assert.New(t)
+	params := map[string]string{"graphType": "versionedApp", "duration": "300s", "idleNodes": "true"}
+	assertGraph(params, assert)
+}
+
+func TestDisplayOperationNodes(t *testing.T) {
+	assert := assert.New(t)
+	params := map[string]string{"graphType": "versionedApp", "duration": "300s", "operationNodes": "true"}
+	assertGraph(params, assert)
+}
+
+func TestDisplayServiceNodes(t *testing.T) {
+	assert := assert.New(t)
+	params := map[string]string{"graphType": "versionedApp", "duration": "300s", "injectServiceNodes": "true"}
+	assertGraph(params, assert)
+}
+func assertGraphInvalid(params map[string]string, assert *assert.Assertions) {
+	params["namespaces"] = utils.BOOKINFO
+	_, statusCode, _ := utils.Graph(params)
+	assert.Equal(statusCode, 400)
+}
+
+func assertGraph(params map[string]string, assert *assert.Assertions) {
+	params["namespaces"] = utils.BOOKINFO
+	graph, statusCode, err := utils.Graph(params)
+	assert.Equal(200, statusCode)
+	assert.Nil(err)
+
+	for key, value := range params {
+		switch key {
+		case "duration":
+			assert.Contains(value, fmt.Sprintf("%d", graph.Duration))
+		case "graphType":
+			assert.Equal(value, graph.GraphType)
+		}
+	}
+	assert.NotNil(graph.Elements.Nodes)
+	assert.NotNil(graph.Elements.Edges)
+	// TODO better way to check if there are any graph nodes at all to be able to verify requested ones
```

Comment:
This is because the test only looks at whether or not nodes exist but doesn't make any assertions about those nodes? e.g. `TestDisplayServiceNodes` doesn't verify that there are service nodes in the graph just that there are nodes in the graph?

Reply:
added wait there